backend/api/Areas/Tools/Helpers/ImportPropertiesHelper.cs

```diff
@@ -46,6 +46,82 @@ public ImportPropertiesHelper(IPimsAdminService pimsAdminService, ILogger logger
         #endregion
 
         #region Methods
+        /// <summary>
+        /// Parse the value to extract the PID number.
+        /// If it's invalid it will return -1.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        private int ParsePid(string value)
+        {
+            if (int.TryParse(value?.Replace("-", ""), out int pid))
+                return pid;
+
+            return -1;
+        }
+
+        /// <summary>
+        /// Delete properties from inventory that match the parcel PID or building name.
+        /// </summary>
+        /// <param name="properties"></param>
+        /// <param name="updatedBefore">Only allow deletes to properties updated before this date.</param>
+        /// <returns></returns>
+        public IEnumerable<Entity.Property> DeleteProperties(IEnumerable<Model.ImportPropertyModel> properties, DateTime? updatedBefore = null)
+        {
+            if (properties == null) throw new ArgumentNullException(nameof(properties));
+
+            var results = new List<Entity.Property>();
+            foreach (var property in properties)
+            {
+                var pid = ParsePid(property.ParcelId ?? property.PID);
```

Comment:
so the PID can be in either of these fields? Actually I guess based on your ordering most of the time the ParcelId is equal to the PID?